src/snowflake/cli/plugins/nativeapp/codegen/sandbox.py

```diff
@@ -194,24 +208,95 @@ def execute_script_in_sandbox(
     elif env_type == ExecutionEnvironmentType.VENV:
         return _execute_in_venv(
-            script_source, kwargs.get("path"), cwd=cwd, timeout=timeout
+            script_source,
+            kwargs.get("path"),
+            cwd=cwd,
+            timeout=timeout,
+            env_vars=env_vars,
         )
     elif env_type == ExecutionEnvironmentType.CONDA:
         return _execute_in_conda_env(
-            script_source, kwargs.get("name"), cwd=cwd, timeout=timeout
+            script_source,
+            kwargs.get("name"),
+            cwd=cwd,
+            timeout=timeout,
+            env_vars=env_vars,
         )
     elif env_type == ExecutionEnvironmentType.SYSTEM_PATH:
-        return _execute_with_system_path_python(script_source, cwd=cwd, timeout=timeout)
+        return _execute_with_system_path_python(
+            script_source, cwd=cwd, timeout=timeout, env_vars=env_vars
+        )
     else:  # ExecutionEnvironmentType.CURRENT
         return _execute_python_interpreter(
-            sys.executable, script_source, cwd=cwd, timeout=timeout
+            sys.executable, script_source, cwd=cwd, timeout=timeout, env_vars=env_vars
         )
+
+
+class SandboxEnvBuilder(EnvBuilder):
+    """
+    A virtual environment builder that can be used to build an environment suitable for
+    executing user-provided python scripts in an isolated sandbox.
+    """
+
+    def __init__(self, path: Path, **kwargs) -> None:
+        """
+        Creates a new builder with the specified destination path. The path need not
+        exist, it will be created when needed (recursively if necessary).
+
+        Parameters:
+            path (Path): The directory in which the sandbox environment will be created.
+        """
+        super().__init__(**kwargs)
+        self.path = path
+        self._context: Any = None  # cached context
+
+    def post_setup(self, context) -> None:
+        self._context = context
+
+    def ensure_created(self) -> None:
+        """
+        Ensures that the sandbox environment has been created and correctly initialized.
+        """
+        if self.path.exists():
+            self._context = self.ensure_directories(self.path)
+        else:
+            self.path.mkdir(parents=True, exist_ok=True)
+            self.create(
+                self.path
+            )  # will set self._context through the post_setup callback
+
+    def run_python(self, *args) -> None:
+        """
+        Executes the python interpreter in the sandboxed environment with the provided arguments.
+        """
+        positional_args = [
+            self._context.env_exe,
+            "-E",  # passing -E ignores all PYTHON* env vars
+            *args,
+        ]
+        kwargs = {
+            "cwd": self._context.env_dir,
+            "stderr": subprocess.STDOUT,
+        }
+        env = dict(os.environ)
+        env["VIRTUAL_ENV"] = self._context.env_dir
+        subprocess.check_output(positional_args, **kwargs)
```

Comment:
Any reason not to return the `str` output?

Reply:
No particular reason, updated.